spark_log_parser/parsing_models/application_model.py

```diff
@@ -310,10 +317,8 @@ def __init__(self, eventlogpath, bucket=None, stdoutpath=None, debug=False):  #
                 self.jobs[0].add_event(line, False)
 
         if not self.cloud_platform:
-            self.cloud_platform = "emr"
-            self.cloud_provider = "aws"
-            # self.spark_metadata['cloud_platform'] = 'emr'
-            # self.spark_metadata['cloud_provider'] = 'aws'
+            raise UrgentEventValidationException(
+                "Unable to determine the platform this job was run on - these logs may be incomplete and missing SparkListenerEnvironmentUpdate events.")
```

Comment:
If we still don't have a `cloud_platform` down here, then it seems likely that the logs are incomplete and missing data.

Reply:
It seems to me that we should be able to determine reliably Databricks vs. EMR vs (future platforms here ...) while parsing logs as opposed to filling in a default down here that may or may not be correct.

Reply:
I agree, I don't like it either. EMR has proved tricky so far due to the variability of log content (See my comment about `EnvironmetUpdate`). It's possible we could use the very first event, which I'm pretty sure is consistent, but it doesn't explicitly identify EMR.